Apps/WebClient/src/ClientApp/src/components/report/ReportsComponent.vue

```diff
@@ -45,6 +46,14 @@ const medicationRequestReport = "medication-request-report";
 const noteReport = "note-report";
 const laboratoryReport = "laboratory-report";
 const hospitalVisitReport = "hospital-visit-report";
+const medicationsText = "Medications";
```

Comment:
The report and the text can be merged into a single object map with the EntryType values this would allow you to loop through the configuration and tests each key against the config.

https://stackoverflow.com/questions/52296113/using-enum-as-object-keys

```JS
{
  [EntryType.Medication]: {
    value: "medication-report",
    text: "Medications"
    }, 
    // etc.
}
```

you could use Object.keys  to loop through reliable values to test against 

`ConfigUtil.isDependentDatasetEnabled(key)`
and
`ConfigUtil.isDatasetEnabled(key)`

This is much easier to maintain.

Reply:
Hi Ray, you can choose to take this or not, but it doesn't seem my comment made was fully understood, I'll add some detail.

My intention was something along the lines of:
```TypeScript
  loadReportOptions(isEnabledPredicate: (entryType: string) => boolean) {

        for (var entryType in EntryType) {
            if (isEnabledPredicate(entryType)) {
                this.reportTypeOptions.push({
                    // This could be where mike's suggestion comes in for the use of the entryTypeMap values or the config option I suggested.
                    // value: reportOptionConfig[entryType].component, or value: entryTypeMap[entryType].reports.componentName
                    // text: reportOptionConfig[entryType].component, or text: entryTypeMap[entryType].reports.optionText
                });
            }
        }
    }

    created(): void {
        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);

        const isEnabledPredicate: (dataSetName: EntryType) => boolean = this.isDependent
        ? ConfigUtil.isDatasetEnabled
        : ConfigUtil.isDependentDatasetEnabled;
        
        this.loadReportOptions(isEnabledPredicate);
    }
```

Reply:
I just went with Mike's suggestion and it works with less changes. However I see what you are suggesting and looks like a very good way to do it.